libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlag {
+    key: String,
+    filters: PostHogLocalEvaluationFlagFilters,
+    active: bool,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilters {
+    groups: Vec<PostHogLocalEvaluationFlagFilterGroup>,
+    multivariate: PostHogLocalEvaluationFlagMultivariate,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterGroup {
+    variant: Option<String>,
+    properties: Option<Vec<PostHogLocalEvaluationFlagFilterProperty>>,
+    rollout_percentage: f64,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterProperty {
+    key: String,
+    value: PostHogFlagFilterPropertyValue,
+    operator: String,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum PostHogFlagFilterPropertyValue {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+    List(Vec<String>),
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariate {
+    variants: Vec<PostHogLocalEvaluationFlagMultivariateVariant>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariateVariant {
+    key: String,
+    rollout_percentage: f64,
+}
+
+pub struct PostHogClient {
+    server_api_key: String,
+    client_api_key: String,
+    project_id: String,
+    private_api_url: String,
+    public_api_url: String,
+    client: Arc<reqwest::Client>,
+}
+
+pub struct FeatureStore {
+    flags: HashMap<String, PostHogLocalEvaluationFlag>,
+}
+
+impl Default for FeatureStore {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl FeatureStore {
+    pub fn new() -> Self {
+        Self {
+            flags: HashMap::new(),
+        }
+    }
+
+    pub fn update_flags(&mut self, flags: Vec<PostHogLocalEvaluationFlag>) {
+        self.flags.clear();
+        for flag in flags {
+            if flag.active {
+                self.flags.insert(flag.key.clone(), flag);
+            }
+        }
+    }
+
+    /// Generate a consistent hash for a user ID (e.g., tenant ID).
+    ///
+    /// The implementation is different from PostHog SDK. In PostHog SDK, it is sha1 of `user_id.distinct_id.salt`.
+    /// However, as we do not upload all of our tenant IDs to PostHog, we do not have the PostHog distinct_id for a
+    /// tenant.
+    fn consistent_hash(user_id: &str) -> f64 {
+        let mut hasher = sha2::Sha256::new();
+        hasher.update(user_id);
+        let hash = hasher.finalize();
+        let hash_int = u64::from_le_bytes(hash[..8].try_into().unwrap());
+        hash_int as f64 / u64::MAX as f64
+    }
+
+    /// Evaluate a condition. Returns `None` if the condition cannot be evaluated due to parsing error or missing
+    /// property.
+    fn evaluate_condition(
+        &self,
+        operator: &str,
+        provided: &PostHogFlagFilterPropertyValue,
+        requested: &PostHogFlagFilterPropertyValue,
+    ) -> Option<bool> {
+        match operator {
+            "exact" => {
```

Comment:
I'm not sure if the performance is a concern for this client, but it's definitely can be improved.

We can parse PostHog's response one time to smth like 'CompiledExpression', where we will store some Enum for operators instead of strings, parsed float values, etc. So we will not parse it every time we evaluate an expression

But it may be done later

Reply:
There will be another thing to cache such responses so that evaluation doesn't need to be done every time the user calls `get_feature`